pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```diff
@@ -334,6 +349,133 @@ public RepositoryData generate() {
         return repackage(new File(repoParentDir, "maven-repository"));
     }
 
+    public RepositoryData resolveAndRepackage() {
+        try {
+            log.info("Generating maven repository");
+            File sourceDir = createMavenGenerationDir();
+            log.info("Source dir " + sourceDir.getAbsolutePath());
+            boolean tempBuild = PigContext.get().isTempBuild();
+            final MavenArtifactResolver mvnResolver = MavenArtifactResolver.builder()
+                    .setUserSettings(new File(FileUtils.getConfiguredIndySettingsXmlPath(tempBuild)))
+                    .setLocalRepository(sourceDir.getAbsolutePath())
+                    .build();
+
+            Map<String, String> params = pigConfiguration.getFlow().getRepositoryGeneration().getParameters();
+            String bomConstraintGroupId = generationData.getBomGroupId();
+            String bomConstraintArtifactId = generationData.getBomArtifactId();
+            String bomConstraintVersion = getRedhatVersion(
+                    new org.jboss.da.model.rest.GAV(
+                            bomConstraintGroupId,
+                            bomConstraintArtifactId,
+                            generationData.getBomVersion()),
+                    tempBuild);
+
+            final Artifact bom = new DefaultArtifact(
+                    bomConstraintGroupId,
+                    bomConstraintArtifactId,
+                    null,
+                    "pom",
+                    bomConstraintVersion);
+            final List<Dependency> bomConstraints = mvnResolver.resolveDescriptor(bom).getManagedDependencies();
```

Comment:
`bomConstraints` is the most important thing, right? The list of dependencies, "roots" of resolving. 
Could you rename it, to `dependencies` maybe?

Reply:
by BomConstraints, I wanted to convey that, this is scope under which artifacts will get resolved. So this is kinda constraint on resolving artifacts.

Reply:
maybe I'm missing something. What is it constraining? What would get resolved if this list was empty?